bundles/config/org.eclipse.smarthome.config.discovery/src/main/java/org/eclipse/smarthome/config/discovery/DiscoveryResult.java

```diff
@@ -80,12 +80,20 @@
      * discovered. Its actual value can be retrieved from the {@link DiscoveryResult#getProperties()} map. Such unique
      * identifiers are typically the <code>ipAddress</code>, the <code>macAddress</code> or the
      * <code>serialNumber</code> of the discovered thing.
-     * 
+     *
      * @return the representation property of this result object (could be null)
      */
     public String getRepresentationProperty();
 
     /**
+     * Returns the actual value of the representation property of the result object.
+     *
+     * @see #getRepresentationProperty()
+     * @return the value of the representation property of this result object (could be null)
+     */
+    public Object getRepresentationValue();
```

Comment:
I am not too fond in having such "convenience" methods as part of the API.
To me, the API is complete as it offers `getRepresentationProperty()` and `getProperties()` - a client can nicely get hold of all information that he needs.
Having such an additional method on the interface rather brings up the risk that this new method returns something different than `getProperties().get(getRepresentationProperty())`.
I'd therefore suggest to move this convenience method to the class where you actually need it (PersistentInbox).